MHVLib/Clients/MHVRemoteMonitoringClient.m

```diff
@@ -18,61 +18,109 @@
 //
 
 #import <Foundation/Foundation.h>
+#import "MHVCommon.h"
 #import "MHVRemoteMonitoringClient.h"
 #import "MHVClient.h"
 #import "MHVJsonSerializer.h"
+#import "MHVConnectionProtocol.h"
+#import "MHVRestRequest.h"
+#import "MHVServiceResponse.h"
+#import "MHVConnectionFactory.h"
+#import "MHVConnection.h"
+
+@interface MHVRemoteMonitoringClient ()
+
+@property (nonatomic, weak) id<MHVConnectionProtocol>     connection;
+
+@end
 
 @implementation MHVRemoteMonitoringClient
 
-+ (NSURLSessionTask*) requestWithPath:(NSString* _Nonnull)path
-                               method:(NSString* _Nonnull)method
-                           pathParams:(NSDictionary * _Nullable)pathParams
-                          queryParams:(NSDictionary* _Nullable)queryParams
-                           formParams:(NSDictionary * _Nullable)formParams
-                                 body:(id _Nullable)body
+- (instancetype)initWithConnection:(id<MHVConnectionProtocol>)connection
+{
+    MHVASSERT_PARAMETER(connection);
+    
+    self = [super init];
+    if (self)
+    {
+        _connection = connection;
+    }
+    
+    return self;
+}
+
+//PENDING: Swagger generated code needs to update to be injected with RemoteMonitoringClient & remove this method
++ (NSURLSessionTask *)requestWithPath:(NSString *_Nonnull)path
+                               method:(NSString *_Nonnull)method
```

Comment:
we should avoid using the term 'method' here. It's important to differentiate the difference between the legacy methods and the new rest 'apis'. The develop documentation on the HealthVault website refers to t these as 'REST API'. This might be something Mike has to do in the swagger gen templates.